test/affinity.cxx

```diff
@@ -0,0 +1,430 @@
+			nextProcessor(processorInfo, info, groupIndex, maskOffset);
+			return {groupIndex, maskOffset};
+		}()
+	};
+
+	auto affinity{substrate::make_unique_nothrow<substrate::affinity_t>(1U)};
+
+	REQUIRE(affinity);
+	REQUIRE(affinity->numProcessors() == 1);
+	REQUIRE(affinity->begin() != affinity->end());
+	REQUIRE(affinity->begin()->first == processor.first);
+	REQUIRE(affinity->begin()->second == processor.second);
+#endif
+}
+
+TEST_CASE("user pinning", "[affinity_t]")
+{
+#if defined(_POSIX_THREADS) && defined(_GNU_SOURCE)
+	const auto processor
+	{
+		[&]() -> uint32_t
+		{
+			cpu_set_t affinitySet{};
+			REQUIRE(sched_getaffinity(0, sizeof(cpu_set_t), &affinitySet) == 0);
+			for (uint32_t i{0}; i < CPU_SETSIZE; ++i)
+			{
+				if (CPU_ISSET(i, &affinitySet))
+					return i;
+			}
+			static_assert(CPU_SETSIZE < UINT32_MAX);
+			return UINT32_MAX;
+		}()
+	};
+	const std::initializer_list<uint32_t> processorList {processor};
+
+	REQUIRE(processor != UINT32_MAX);
+	auto affinity{substrate::make_unique_nothrow<substrate::affinity_t>(0U, processorList)};
+
+	REQUIRE(affinity);
+	REQUIRE(affinity->numProcessors() == 1);
+	REQUIRE(affinity->begin() != affinity->end());
+	REQUIRE(*affinity->begin() == processor);
+#elif defined(_POSIX_THREADS) && defined(__APPLE__)
+	std::default_random_engine gen{std::random_device{}()};
+	std::uniform_int_distribution<uint32_t> distrib{0U, std::thread::hardware_concurrency() - 1U};
+	const uint32_t processor {distrib(gen)};
+	const std::initializer_list<uint32_t> processorList {processor};
+
+	REQUIRE(processor != UINT32_MAX);
+
+	auto affinity{substrate::make_unique_nothrow<substrate::affinity_t>(0U, processorList)};
+
+	REQUIRE(affinity);
+	REQUIRE(affinity->numProcessors() == 1);
+	REQUIRE(affinity->begin() != affinity->end());
+	REQUIRE(*affinity->begin() == processor);
+#elif defined(_WIN32)
+	const auto processor
+	{
+		[&]() -> std::pair<uint16_t, uint8_t>
+		{
+			const auto processorInfo{retrieveProcessorInfo()};
+			auto info{processorInfo.begin()};
+			uint16_t groupIndex{};
+			uint8_t maskOffset{};
+			nextProcessor(processorInfo, info, groupIndex, maskOffset);
+			return {groupIndex, maskOffset};
+		}()
+	};
+	const std::initializer_list<uint32_t> processorList {0U};
+
+	auto affinity{substrate::make_unique_nothrow<substrate::affinity_t>(0U, processorList)};
+
+	REQUIRE(affinity);
+	REQUIRE(affinity->numProcessors() == 1);
+	REQUIRE(affinity->begin() != affinity->end());
+	REQUIRE(affinity->begin()->first == processor.first);
+	REQUIRE(affinity->begin()->second == processor.second);
+#endif
+}
+
+TEST_CASE("pin second core", "[affinity_t]")
+{
+#if defined(_POSIX_THREADS) && defined(_GNU_SOURCE)
+	const auto processor
+		{
+			[&]() -> uint32_t
+			{
+				cpu_set_t affinitySet{};
+				REQUIRE(sched_getaffinity(0, sizeof(cpu_set_t), &affinitySet) == 0);
+				if (CPU_COUNT(&affinitySet) < 2)
+					throw std::runtime_error("This test can only be run on a multi-core machine with at "
```

Comment:
This is a good check and works, but we wonder as a result if there's something in Catch2 we can call to mark the test skipped or expected-fail?

Reply:
Not in Catch2: https://www.reddit.com/r/cpp/comments/10jdm0o/catch2_added_skip_and_more_builtin_matchers/

Reply:
Ah, a shame but that kinda figures. Thank you